### DISCUSSION TITLE
Questions on first experiments with the modeler

### DISCUSSION THREAD

**Discussion by spacejaco** [ThalesGroup/xsmp-modeler-core#18] [Q&A]
Hello. I work for Spacebel and I have to analyse the capabilities of XSMP.
1/ I have succeeded in installing the modeler, except the error: C/C++ indexer has encountered a problem; Cannot invoke "java.util.List.iterator()" because "org.eclipse.cdt.managedbuilder.core.ManagedBuildManager.projectTypes" is null"
Do you know how to solve that ?
2/ I have succeeded in creating a xsmpcat, based on the PPT of a presentation you have made. It works quite fine, with interesting capabilities. It generates smpcat's and some C++ code. However I don't manage to generate a working test: something is missing related to TAS CompileChainNature and PythonNature.
3/ It could be very useful to have a very simple example (for example a model that issues a log through ILogger) running from A to Z (up to a running application)
4/ It is not clear how to use the generated code together with an assembly (for example with just 1 instance of the model proposed at point 3/)
5/ I know that the question was issued in the past, but do you have a way to convert a (simple) smpcat into xsmpcat, the underlying goal is to have an example starting from typical learning material we use at Spacebel.
Thanks

**Comment by ydaveluy** [comment]
Hello @spacejaco 

1/2/ : It seems that you are using an old version of XSMP with TAS specific stuffs. 

3/ : The actual version (1.1.0-rc1) does not provide a full working profile. We are working on a specific one for the SIMSAT framework #13 that will be available in rc2.

4/ : The modeler do not provides tools for assembly. Only catalogue related stuff are handled.

5/ : It is possible to convert an smpcat file to xsmpcat --> right click on the smpcat file --> convert to xsmpcat. 

If you have the SIMSAT framework, I encourage you to test the branch #13 : `mvn install` then install the plugins from the archive generated in org.eclipse.xsmp.repository/target/org.eclipse.xsmp.repository.zip


Yannick
